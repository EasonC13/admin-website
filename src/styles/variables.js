const variables = {
  black: "#000000",
  white: "#ffffff",
  brandBlue: "#0099cc",
  brandDarkBlue: "#005f7f",
  brandOrange: "#ff9933",
  brandDarkOrange: "#ff6a33",
  lightGrey: "#e5e5e5",
  greyBlue: "#c1c9d1",
  brandDark: "#324353",
  faintBlue: "#f5f8fb",
  faintGreen: "#f5fbf7",

  headerUiBg: "#f3f8fc",
  lowTranslucencyWhite: "rgba(255, 255, 255, 0.7)",

  successColor: "#28a745",
  warningColor: "#ffbf00",
  warningBackground: "#fff9e8",
  invalidColor: "#e46767",
  invalidBackground: "#fbebe7"
};

export default variables;
