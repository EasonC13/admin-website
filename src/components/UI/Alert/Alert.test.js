// Button.spec.js
import React from "react";
import enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { matchers } from "jest-emotion";
import renderer from "react-test-renderer";
import { Alert, RedAlert } from ".";
import { invalidBackground, invalidColor } from "../../../styles/variables";

expect.extend(matchers);

describe("Alert component", () => {
  enzyme.configure({ adapter: new Adapter() });
  it("should be defined", () => {
    expect(Alert).toBeDefined();
  });

  it("shows alert text", () => {
    const alert = renderer.create(<Alert>Test Alert</Alert>);
    expect(alert.toJSON()).toMatchSnapshot();
  });

  it("shows red alert", () => {
    const alert = renderer.create(<RedAlert>Test Alert</RedAlert>);
    const tree = alert.toJSON();
    expect(tree).toHaveStyleRule("background", invalidBackground);
    expect(tree).toHaveStyleRule("color", invalidColor);
  });
});
