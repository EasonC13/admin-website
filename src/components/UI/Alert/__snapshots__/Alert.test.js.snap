// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component shows alert text 1`] = `
<div
  className="css-1d2iil2-Alert"
>
  Test Alert
</div>
`;
